include/private/tdymeshimpl.h

```diff
@@ -170,18 +179,114 @@ typedef struct TDyMesh {
   PetscInt   num_faces;
   PetscInt   num_edges;
   PetscInt   num_vertices;
-  PetscInt num_boundary_faces;
+  PetscInt   num_boundary_faces;
   PetscInt   num_subcells;
 
+  PetscInt max_vertex_cells, max_vertex_faces;
+
   TDyCell    cells;
   TDySubcell subcells;
   TDyVertex  vertices;
   TDyEdge    edges;
   TDyFace    faces;
 
   TDyRegion region_connected;
+
+  PetscInt *closureSize, **closure, maxClosureSize;
 } TDyMesh;
 
-PETSC_INTERN PetscErrorCode TDyBuildMesh(TDy);
-PETSC_INTERN PetscErrorCode TDyAllocateMemoryForMesh(TDy);
+PETSC_INTERN PetscErrorCode TDyMeshCreate(DM,PetscReal*,PetscReal*,PetscReal*,TDyMesh**);
+PETSC_INTERN PetscErrorCode TDyMeshDestroy(TDyMesh*);
+
+// These don't work, and we'll likely get rid of them.
+//PETSC_INTERN PetscErrorCode TDyMeshReadGeometry(TDyMesh*,const char*);
+//PETSC_INTERN PetscErrorCode TDyMeshWriteGeometry(TDyMesh*,const char*);
```

Comment:
Is there a new issue to add this functionality back in the future?

Reply:
It depends on whether we need it. Is this just for saving time in debugging? At this point, we could make a proper effort to optimize the calculation of geometry so we don't need this feature.